Conventional/Conventions/Cecil/MustNotResolveCurrentTimeViaDateTimeDirectlyConventionSpecification.cs

```diff
@@ -1,38 +1,46 @@
-﻿using System;
-using System.Linq;
-using Conventional.Conventions;
-using Mono.Cecil;
-using Mono.Cecil.Cil;
-
-namespace Conventional.Cecil.Conventions
-{
-    public class MustNotUseDateTimeNowConventionSpecification : ConventionSpecification
-    {
-        protected override string FailureMessage
-        {
-            get { return "Must not use DateTime.Now. It is called {0} times in this type."; }
-        }
-
-        public override ConventionResult IsSatisfiedBy(Type type)
-        {
-            var nowAssignments =
-                type.ToTypeDefinition()
-                    .Methods
-                    .Where(method => method.HasBody)
-                    .SelectMany(method => method.Body.Instructions)
-                    .Where(
-                        x =>
-                            x.OpCode == OpCodes.Call && x.Operand is MethodReference &&
-                            ((MethodReference) x.Operand).DeclaringType.FullName == "System.DateTime" &&
-                            ((MethodReference) x.Operand).Name == "get_Now")
-                    .ToArray();
-
-            if (nowAssignments.Any())
-            {
-                return ConventionResult.NotSatisfied(type.FullName, FailureMessage.FormatWith(nowAssignments.Count()));
-            }
-
-            return ConventionResult.Satisfied(type.FullName);
-        }
-    }
+﻿using System;
+using System.Linq;
+using Conventional.Conventions;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+namespace Conventional.Cecil.Conventions
+{
```

Comment:
I agree the name you prefer is more concise. I considered it myself but finally decided against it. MustNotResolveCurrentTimeViaDateTimeDirectlyConventionSpecification I chose attempts to clearly indicate the convention is not to avoid *any* usage of DateTime (like constructing the type with new DateTime(2014, 10,10) or using one of its instance methods on such) but rather only those parts of it that "bind" current time. What do you think? Do you have any other idea that would address both accuracy and conciseness? Either way I am leaving the final decision up to you of course.